packages/starlight/index.ts

```diff
@@ -78,6 +82,13 @@ export default function StarlightIntegration({
 					scopedStyleStrategy: 'where',
 					// If not already configured, default to prefetching all links on hover.
 					prefetch: config.prefetch ?? { prefetchAll: true },
+					experimental: {
+						// For projects in server mode enable global route priority to allow custom
+						// routes to customize any path.
+						...(config.output === 'server' && {
+							globalRoutePriority: true,
+						}),
+					}
```

Comment:
Could you say a bit more about what does and doesn’t work with and without this?

Reply:
Since everything will be pre-rendered basically everything works as intended for most use cases.

The only difference is in the case of a pre-rendered custom route generating the same path as a docs page.

Ideally, that should be an error or at least a warning (tracked on https://github.com/withastro/astro/issues/9832), but it currently is silently ignored I believe it is better to have the custom page overriding the docs page instead of the opposite.

Reply:
Thanks for the clarification! I’m thinking about it. I agree with the basic premise that “it is better to have the [user’s] custom page overriding the docs page instead of the opposite”.

The only thing I’m wondering about is if setting this experimental mode invisibly could mess with a user’s wider project in other unexpected ways, which is something we need to consider given this applies globally to all their routes. Have you seen this option cause any issues for folks so far?

Maybe a good compromise is to set this _only_ if the user themselves didn’t already set it:

```suggestion
					experimental: {
						// For projects in server mode enable global route priority to allow custom
						// routes to customize any path.
						...(config.output === 'server' && {
							globalRoutePriority: config.experimental.globalRoutePriority ?? true,
						}),
					}
```

That way we can default to the new experimental behaviour, but still let people opt out if they need.

Reply:
OK, so important question from above is:

> Have you seen this option cause any issues for folks so far?

(or are there obvious potential issues you can imagine?)

That can inform whether we set this for folks or let them opt in to it themselves if they need it.

Reply:
We can leave this out for now... Docs page will win in case of conflicts.

It is a requirement for the other PR that makes Starlight pages SSR since then it would override any custom route.

I hope cases of conflicts will have at least a warning in the future anyway, so I don't really mind which one wins here that much.

Reply:
Let users opt-in if they need/want it.

Reply:
Just so I don't leave the question open

> Have you seen this option cause any issues for folks so far?

No, I haven't. In fact I haven't seen anyone even mentioning that they are using it, so the only use case I know of is mine and it is working fine for me.

Reply:
Ok, here’s a radical proposal, how about we straight up enable this for all projects?

```suggestion
					experimental: {
						globalRoutePriority: true,
					},
```

This will give us some more data about if it causes issues for people and prepare for when it becomes the default priority algorithm in a future Astro major.

Couple of notes:
- We’ll also need to bump the minimum supported Astro version to 4.2 in this case in the peer dependencies of `packages/starlight`
- We should probably add a couple of extra changesets specifically for these two things for visibility (new minimum version support and enabling the `globalRoutePriority` flag)

Reply:
> Also basically makes the recently added disable404Route option obsolete I think? 😁

I don't think it does, both would be static routes. Neither would be guaranteed to win, and I'm not sure which _would_ win. It would also generate a warning about the conflict.

---

But I agree about enabling it for everyone.

Reply:
Oh right, we still have two `404.astro` routes in that case. We still would need a way for Starlight to mark it's default one as lower priority.

Reply:
Changed to enable it always and added a changeset for it